buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -89,6 +89,9 @@
  */
 @UnstableApi
 final class AdaptivePoolingAllocator {
+    private static final int LOW_MEM_THRESHOLD = 512 * 1024 * 1024;
+    private static final boolean IS_LOW_MEM = Runtime.getRuntime().maxMemory() <= LOW_MEM_THRESHOLD;
```

Comment:
We already capture this info in `PlatformDependent0` IIRC

Reply:
We don't. The PlatformDependents do heuristic dances to try and figure out the max _direct_ memory, though. But here we're interested in the heap memory.